### PR TITLE
Add main components

### PR DESCRIPTION
**Pull request by anas599**
- Add Numbers buttons
- Add mathematical operators signs
- Add styling

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Anas, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure that your ReadMe is professionally documented, and adhere(s) to the following standard(s):
- [ ] Kindly ensure that provided placeholder texts/ guide instructions are removed and/or replaced with appropriate contents applicable under each section of your ReadMe. Similarly, kindly also remove guide notes for _license_ creation under the _license_ section.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/calculator.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+function mainCalculator() {
```
Comment (rloterh) [inline src/components/calculator.js]: -  [ ] In this activity, we want the _Calculator_ component to be class-based component. Kindly improve upon this aspect.

File: src/App.js [hunk 2]
```diff
@@ -1,29 +1,4 @@
-import logo from './logo.svg';
 import './App.css';
+import mainCalculator from './components/calculator';
```
Comment (rloterh) [inline src/App.js:2]: - [ ] Similarly, let's make the main component, _App_ component a class-based component. The _App_ component should be similar to:

```javascript
// src/App.js

import React, { Component } from 'react';
import Calculator from ./components; // Import Calculator component

class App extends Component {
  render() {
    // reference as a js object
    return {
      <Calculator />
    };
  }
}
```

File: README.md [hunk 3]
```diff
@@ -66,20 +65,17 @@ Video presenting the features the Project
...
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- LIVE DEMO -->
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-> Add a link to your deployed project.
-
 - [Live Demo Link](https://anas599.github.io)
```
Comment (rloterh) [inline README.md]: - [x] Kindly endeavor to remove/update redundant demo link.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Anas, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦   🏃   

## Highlights 👍  
- Good job so far with component creation!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rloterh** [comment]
Hi @anas599 

Kindly ignore current requirements in  the given [project requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md) for this activity. Let's say, it's in the process of being updated! (Apologize for any inconveniences caused!)

Simply update your components to a class-based components and then resubmit for another review. 

Cheers! 😃

**Comment by anas599** [comment]
> 

Hi @rloterh thanks for the update 
I'm having one issue though when I do it as you described it shows eslint error and to fix it I get a suggestion to convert it to a function (which is back to the original one I submitted)
Kindly provide me with what requirements I should follow,
Screenshot 2023-02-13 at 10 37 34 PM


Regards,

**Comment by rloterh** [comment]
Hi @anas599 

Please try something like the ff:

```javascript
import React from 'react';

class Calculator extends React.Component {
  ...
  ...
  ...
}

export default Calculator;

```

Cheers! 😃

**Comment by anas599** [comment]
> Hi @anas599
> 
> Please try something like the ff:
> 
> ```js
> import React from 'react';
> 
> class Calculator extends React.Component {
>   ...
>   ...
>   ...
> }
> 
> export default Calculator;
> ```
> 
> Cheers! 😃

Hi @rloterh  Could you please be more specific as I'm getting this error from eslint even if I implement the class 
Screenshot 2023-02-13 at 11 48 47 PM

**Comment by anas599** [comment]
> > Hi @anas599
> > Please try something like the ff:
> > ```js
> > import React from 'react';
> > 
> > class Calculator extends React.Component {
> >   ...
> >   ...
> >   ...
> > }
> > 
> > export default Calculator;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Cheers! 😃
> 
> Hi @rloterh Could you please be more specific as I'm getting this error from eslint even if I implement the class Screenshot 2023-02-13 at 11 48 47 PM

**Comment by rloterh** [comment]
Hi @anas599 
I cloned your project project with the below code snippet, without eslint promptings:

```javascript
import React, { Component } from 'react';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      ...
    );
  }
}

export default Calculator;
```

The main component should be similar to:

```javascript
import React from 'react';

import Calculator from './components/Calculator';

class App extends React.PureComponent {
  render() {
    return (
      ...
    );
  }
}

export default App;
```

**Comment by anas599** [comment]
> Hi @anas599 I cloned your project project with the below code snippet, without eslint promptings:
> 
> ```js
> import React, { Component } from 'react';
> 
> class Calculator extends Component {
>   constructor(props) {
>     super(props);
>     this.state = {
>     };
>   }
> 
>   render() {
>     return (
>       ...
>     );
>   }
> }
> 
> export default Calculator;
> ```
> 
> The main component should be similar to:
> 
> ```js
> import React from 'react';
> 
> import Calculator from './components/Calculator';
> 
> class App extends React.PureComponent {
>   render() {
>     return (
>       ...
>     );
>   }
> }
> 
> export default App;
> ```

@rloterh Thanks a lot for your help

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @anas599, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Has a proper Pull Request. ✔️
- No Linter errors ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**